TriblerGUI/widgets/trustpage/js/legend.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Draw the legend based on the current visible network.
+ * Handle clicks to show and hide the legend.
+ */
+
+/**
+ * Draw the legend based on the quantitative data of the graph currently visible
+ * @param data the data of the graph
+ */
+function drawLegend (data) {
```

Comment:
This function is nearly 100 lines long. Is there a way to split it up in different functions so that the code becomes more readable?